File: NuevoComienzo/Curls/decode_backdoor.py
import re

# Las cadenas de codificación del backdoor
p = "aRfguQwJvbc#h l)[W2jkm/op?N!rVisGq=@tB3O4CdeF9n0U(HI_6X]KLM>,DE.-Pxyz15YZ<78AST"
l = "gJvbUWc#,Dh5w*sGq=@tB[!39nO4CuYdeF0a]Kl)iQxZ<7yf2jkSTrVm/oR-Pp?NI_6XLM>E.z18A(H"

def decode(encoded):
    reversed_str = encoded[::-1]
    decoded = ""
    for char in reversed_str:
        if char in l:
            index = l.index(char)
            decoded += p[index]
        else:
            decoded += char
    return decoded

# Probar con algunas cadenas codificadas del backdoor
test_strings = [
    ")HS(*39YUF?",
    "gYhCge*3y3H*Zd3#",
    "3]hZX3C_*Zx*CZxYo*PC3xgss3CCgdZp",
    "whygJ9(*I*9dsXyxa*I*-D!z9(*sZXygp-Dz*HQ(S*I*fMf@*]wbYCX93h"
]

print("=== DECODIFICACIÓN DEL BACKDOOR ===")
for encoded in test_strings:
    try:
        decoded = decode(encoded)
        print(f"Codificado: {encoded}")
        print(f"Decodificado: {decoded}")
        print("---")
    except Exception as e:
        print(f"Error decodificando {encoded}: {e}")

File: NuevoComienzoLinux/wgets/decode_backdoor.js
function decodeTomcatStrings() {
    const p = "aRfguQwJvbc#h l)[W2jkm/op?N!rVisGq=@tB3O4CdeF9n0U(HI_6X]KLM>,DE.-Pxyz15YZ<78AST";
    const l = "gJvbUWc#,Dh5w*sGq=@tB[!39nO4CuYdeF0a]Kl)iQxZ<7yf2jkSTrVm/oR-Pp?NI_6XLM>E.z18A(H";
    
    function f(D) {
        var F = D.split("").reverse();
        var E = "";
        for(let cont = 0; cont < F.length; cont++) {
            for(let cont2 = 0; cont2 < l.length; cont2++) {
                if(l[cont2] == F[cont]) {
                    E = E + "" + p[cont2];
                    break;
                }
            }
        }
        return E;
    }
    
    // Strings importantes del backdoor
    const importantStrings = [
        ")HS(*39YUF?",
        "gYhCge*3y3H*Zd3#",
        "3]hZX3C_*Zx*CZxYo*PC3xgss3CCgdZp",
        "whygJ9(*I*9dsXyxa*I*-D!z9(*sZXygp-Dz*HQ(S*I*fMf@*]wbYCX93h"
    ];
    
    importantStrings.forEach(str => {
        console.log(`Original: ${str} -> Decoded: ${f(str)}`);
    });
}

decodeTomcatStrings();
